src/pages/library/dataBrowser-utils.ts

```diff
@@ -76,98 +96,110 @@ export const getDataModalityListFromDataset = dataset => _.flow(
-      [access === datasetAccessTypes.PENDING, () => div({ style: { color: commonStyles.access.pending, display: 'flex', alignItems: 'center' } }, [
+      [access === datasetAccessTypes.Pending, () => div({ style: { color: commonStyles.access.pending, display: 'flex', alignItems: 'center' } }, [
         icon('lock'),
         div({ style: textStyle }, ['Pending Access'])
       ])],
-      [access === datasetAccessTypes.EXTERNAL, () => h(ButtonOutline, {
+      [access === datasetAccessTypes.External, () => h(ButtonOutline, {
         style: buttonStyle,
         href: dataset['dcat:accessURL'], target: '_blank'
-      }, [div({ style: { fontSize: 12 } }, ['Externally managed']), icon('pop-out', { style: { marginLeft: 10 }, size: 16 })])],
+      }, [
+        div({ style: { fontSize: 12 } }, ['Externally managed']),
+        // @ts-expect-error
+        icon('pop-out', { style: { marginLeft: 10 }, size: 16 })
+      ])],
       [Utils.DEFAULT, () => div({ style: { color: commonStyles.access.granted, display: 'flex', alignItems: 'center' } }, [
         icon('unlock'),
         div({ style: textStyle }, ['Granted Access'])
       ])]),
-    !!requestingAccess && h(RequestDatasetAccessModal, {
+    requestingAccess && h(RequestDatasetAccessModal, {
       datasets: [dataset],
-      onDismiss: () => setRequestingAccess()
+      onDismiss: () => setRequestingAccess(false)
     })
   ])
 }
 
 // These are used to match against by the filter
-const extractTags = dataset => {
+const extractTags = (dataset: DatasetResponse): Tags => {
   return {
     itemsType: 'AttributeValue',
     items: _.flow(_.flatten, _.toLower)([
       getDatasetAccessType(dataset),
-      getConsortiumsFromDataset(dataset),
-      dataset.samples?.genus,
+      getConsortiumTitlesFromDataset(dataset),
+      dataset.samples?.species,
       dataset.samples?.disease,
       getAssayCategoryListFromDataset(dataset),
       getDataModalityListFromDataset(dataset),
-      _.map('dcat:mediaType', dataset.files),
+      _.map('TerraCore:hasFileFormat', dataset.fileAggregate),
       dataset['TerraDCAT_ap:hasDataUsePermission']
-    ])
+    ]) as string
   }
 }
 
-export const prepareDatasetsForDisplay = (datasets, dataCollectionsToInclude) => {
+
+export const prepareDatasetsForDisplay = (datasets: DatasetResponse[], dataCollectionsToInclude: string[]): Dataset[] => {
   const filteredDatasets = _.filter(dataCollectionsToInclude ?
     dataset => _.intersection(dataCollectionsToInclude, _.map('dct:title', dataset['TerraDCAT_ap:hasDataCollection'])).length > 0 :
     _.constant(true),
   datasets)
   return _.map(dataset => {
-    return _.set(['tags'], extractTags(dataset), dataset)
+    return { ...dataset, tags: extractTags(dataset) } as Dataset
   }, filteredDatasets)
 }
 
-export const useDataCatalog = () => {
+interface DataCatalog { dataCatalog: Dataset[]; refresh: () => void; loading: boolean }
```

Comment:
Is the convention to not wrap an interface across lines if it only has a few fields? I would expect this to be written as

```suggestion
interface DataCatalog {
  dataCatalog: Dataset[];
  refresh: () => void;
  loading: boolean
}
```

Reply:
Also, what about using `readonly` for read-only fields? I think `refresh()` is readonly, not sure about `dataCatalog` and `loading`.

Reply:
Sure! I can try both of those